common/src/main/java/com/wynntils/models/seaskipper/SeaskipperModel.java

```diff
@@ -0,0 +1,202 @@
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.components.Model;
+import com.wynntils.core.components.Models;
+import com.wynntils.core.net.Download;
+import com.wynntils.core.net.UrlId;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.mc.event.ContainerSetSlotEvent;
+import com.wynntils.mc.event.MenuEvent;
+import com.wynntils.mc.event.ScreenOpenedEvent;
+import com.wynntils.models.containers.ContainerModel;
+import com.wynntils.models.items.items.gui.SeaskipperDestinationItem;
+import com.wynntils.models.map.pois.SeaskipperPoi;
+import com.wynntils.utils.mc.ComponentUtils;
+import com.wynntils.utils.wynn.ContainerUtils;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import org.lwjgl.glfw.GLFW;
+
+public final class SeaskipperModel extends Model {
+    private static final StyledText OAK_BOAT_NAME = StyledText.fromString("§bOak Boat");
+    private static final Pattern SEASKIPPER_PASS_PATTERN = Pattern.compile("^§b(.*) Pass §7for §b(\\d+)²$");
+
+    private static AbstractContainerScreen actualSeaskipperScreen;
+
+    private final Map<SeaskipperDestinationItem, Integer> destinations = new HashMap<>();
+    private final List<SeaskipperPoi> seaskipperPois = new ArrayList<>();
+    private final List<String> availableNames = new ArrayList<>();
+
+    private List<SeaskipperPoi> availableDestinations = new ArrayList<>();
+    private SeaskipperPoi currentPoi;
+    private int boatSlot;
+    private int containerId = -2;
+
+    public SeaskipperModel(ContainerModel containerModel) {
+        super(List.of(containerModel));
+    }
+
+    @SubscribeEvent
+    public void onScreenOpened(ScreenOpenedEvent.Post event) {
+        if (Models.Container.isSeaskipper(
+                ComponentUtils.getUnformatted(event.getScreen().getTitle()))) {
+            actualSeaskipperScreen = (AbstractContainerScreen) event.getScreen();
+
+            loadSeaskipperPois();
+        }
+    }
+
+    @SubscribeEvent
+    public void onMenuOpened(MenuEvent.MenuOpenedEvent event) {
+        if (Models.Container.isSeaskipper(ComponentUtils.getUnformatted(event.getTitle()))) {
+            containerId = event.getContainerId();
+        }
+    }
+
+    public void closeScreen() {
+        ContainerUtils.closeContainer(actualSeaskipperScreen.getMenu().containerId);
+    }
+
+    @SubscribeEvent
+    public void onSetSlot(ContainerSetSlotEvent event) {
+        if (event.getContainerId() != containerId) return;
+
+        if (StyledText.fromComponent(event.getItemStack().getHoverName()).equals(OAK_BOAT_NAME)) {
+            setBoatSlot(event.getSlot());
+            return;
+        }
+
+        Matcher matcher =
```

Comment:
You don't actually have to do all this parsing. I did not notice before, but the SeaskipperDestinationItem is already set up for you.

You just need to do like 
```
        Optional<SeaskipperDestinationItem> seaskipperItemOpt = Models.Item.asWynnItem(itemStack, SeaskipperDestinationItem.class);
```
and check if the optional is present.